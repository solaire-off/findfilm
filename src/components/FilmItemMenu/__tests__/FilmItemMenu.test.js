import React from "react";
import { screen, act, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { FilmItemMenu } from "..";

const mockAddModal = jest.fn();
const mockDeleteModal = jest.fn();

const filmCardActions = [
  {
    name: "Edit",
    callback: mockAddModal,
  },
  {
    name: "Delete",
    callback: mockDeleteModal,
  },
];

it("Current render", async () => {
  await act(async () => {
    render(<FilmItemMenu id={1} actions={filmCardActions} />);
  });
  expect(document.body).toMatchSnapshot();
});

it("Menu items callbacks is work", async () => {
  await act(async () => {
    render(<FilmItemMenu id={1} actions={filmCardActions} />);
  });

  filmCardActions.forEach((item) => {
    userEvent.click(screen.getByText(item.name));
  });

  filmCardActions.forEach((item) => {
    expect(item.callback.mock.calls.length).toEqual(1);
  });
});
