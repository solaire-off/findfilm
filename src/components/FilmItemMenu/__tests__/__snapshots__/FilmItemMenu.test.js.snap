// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current render 1`] = `
<body>
  <div>
    <div
      class="film-card__menu null"
    >
      <button
        aria-label="Film menu"
        class="film-card__menu-toggle"
        type="button"
      />
      <div
        class="film-card__actions"
      >
        <button
          aria-label="Close"
          class="film-card__actions-toggle"
          type="button"
        />
        <button
          class="film-card__actions-item"
          type="button"
        >
          Edit
        </button>
        <button
          class="film-card__actions-item"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</body>
`;
